server/mailet/rate-limiter/src/main/scala/org/apache/james/transport/mailets/EntityType.scala

```diff
@@ -48,62 +49,48 @@ object ConfigurationOps {
     def getOptionalSize(key: String): Option[org.apache.james.util.Size] = mailetConfig.getOptionalString(key)
       .map {
         case "" => throw new IllegalArgumentException(s"'$key' field cannot be empty if specified")
-        case s => s
+        case s  => s
       }
       .map(org.apache.james.util.Size.parse)
   }
 }
 
+object EntityType {
+  def extractRules(entityType: EntityType, duration: Duration, mailetConfig: MailetConfig): Option[Rules] = (entityType match {
```

Comment:
is it expected that adding operations (new functions) will be more frequent than adding new entity types ?

Reply:
Good question (I don't have the answer). What do you think @chibenwa @Arsnael ?

I would also add that if you put this concern aside, the code is usually more cohesive with fp style.

Reply:
> Good question (I don't have the answer). What do you think @chibenwa @Arsnael ?

I expect it to be mostly table but maybe more entity types (HeaderCount? Maybe more inventive people that me can find new ideas?)

Reply:
> Good question (I don't have the answer). What do you think @chibenwa @Arsnael ?

I don't really have a proper answer either but I'm ok with the change